src/climate_indices/compute.py

```diff
@@ -743,3 +743,267 @@ def transform_fitted_gamma(
+    locs = np.zeros((time_steps_per_year,))
+
+    # compute the probability of zero and Pearson
+    # parameters for each calendar time step
+    # TODO vectorize the below loop? create a @numba.vectorize() ufunc
+    #  for application over the second axis
+    for time_step_index in range(time_steps_per_year):
+
+        # get the values for the current calendar time step
+        time_step_values = calibration_values[:, time_step_index]
+
+        # count the number of zeros and valid (non-missing/non-NaN) values
+        number_of_zeros, number_of_non_missing = \
+            utils.count_zeros_and_non_missings(time_step_values)
+
+        # make sure we have at least four values that are both non-missing (i.e. non-NaN)
+        # and non-zero, otherwise use the entire period of record
+        if (number_of_non_missing - number_of_zeros) < 4:
+
+            # we can't proceed, bail out using zeros
+            continue
+
+        # calculate the probability of zero for the calendar time step
+        probability_of_zero = 0.0
+        if number_of_zeros > 0:
+
+            probability_of_zero = number_of_zeros / number_of_non_missing
+
+        # get the estimated parameters, if we have
+        # more than three non-missing/non-zero values
+        if (number_of_non_missing - number_of_zeros) > 3:
+
+            # get the fisk parameters for this time
+            # step's values within the calibration period
+            c[time_step_index], scales[time_step_index], locs[time_step_index], = \
+                scipy.stats.fisk.fit(time_step_values)
+            probabilities_of_zero[time_step_index] = probability_of_zero
+
+    return probabilities_of_zero, c, scales, locs
+
+
+# ------------------------------------------------------------------------------
+@numba.jit
+def _fisk_fit(
+        values: np.ndarray,
+        probabilities_of_zero: np.ndarray,
+        c: np.ndarray,
+        scale: np.ndarray,
+        loc: np.ndarray,
+) -> np.ndarray:
+    """
+    Perform fitting of an array of values to a log-logistic/fisk distribution
+    as described by the fisk parameters and probability of zero arguments.
+
+    :param values: an array of values to fit to the fisk
+        distribution described by the c, loc, and scale
+    :param probabilities_of_zero: probability that the value is zero
+    :param c: first fisk parameter, the shape of the distribution
+    :param scale: second fisk parameter, the scale of the distribution
+    :param loc: third fisk parameter, the location of the distribution
+    """
+
+    # only fit to the distribution if the values array is valid/not missing
+    if not np.all(np.isnan(values)):
+
+        # minimums_possible = _minimum_possible(c, loc, scale)
+        # minimums_mask = values <= minimums_possible
+        zero_mask = np.logical_and((values < 0.0005), (probabilities_of_zero > 0.0))
+        trace_mask = np.logical_and((values < 0.0005), (probabilities_of_zero <= 0.0))
```

Comment:
So, should these go away? Furthermore, if probabilities_of_zero isn't used, it should get taken out of the call signature here, and the `fisk_params()` should be updated to not return it.